File: manual/data/python/src/function_funcy_merge.py
from funcy import merge

class DictMerger:
    def combine_dicts(self, dict1: str, dict2: str) -> str:
        try:
            parsed_dict1 = eval(dict1)
            parsed_dict2 = eval(dict2)
            if not isinstance(parsed_dict1, dict) or not isinstance(parsed_dict2, dict):
                return "Error: input is not a dictionary"
            return str(merge(parsed_dict1, parsed_dict2))
        except Exception:
            return "Error: invalid input"

File: data/cpp/src/function_funcy_merge.cpp
#include <string>
#include <nlohmann/json.hpp>

class DictMerger {
public:
    std::string combine_dicts(const std::string& dict1, const std::string& dict2) {
        try {
           
            auto parsed_dict1 = nlohmann::json::parse(dict1);
            auto parsed_dict2 = nlohmann::json::parse(dict2);

            if (!parsed_dict1.is_object() || !parsed_dict2.is_object()) {
                return "Error: input is not a dictionary";
            }

            for (auto& [key, value] : parsed_dict2.items()) {
                parsed_dict1[key] = value;
            }

            return parsed_dict1.dump();
        } catch (...) {
            return "Error: invalid input";
        }
    }
};